bzt/modules/monitoring.py

```diff
@@ -359,6 +379,265 @@ def __get_cpu_percent(self):
+        return max_cpu_usage
+
+    def _get_cgroups2_max_cpu_usage(self) -> Optional[int]:
+        """
+        Returns maximum possible cpu usage per one second determined from cgroups2 limits. Returns None if there is
+        no cgroups2 limit set.
+        """
+        path = os.path.join(self._cgroup_fs_path, 'cpu.max')
+        if not os.path.exists(path) or not os.path.isfile(path):
+            self._log.warning(f'Unable to determine cgroups2 max cpu usage. File {path} not found')
+            return None
+
+        max_cpu_p = re.compile(r'^([^ ]+)\s([^ ]+)$')
+        try:
+            with open(path) as f:
+                content = f.readline()
+        except IOError:
+            self._log.warning(f'Unable to determine cgroups2 max cpu usage. File {path} is not readable')
+            return None
+        m = max_cpu_p.match(content)
+        if m:
+            try:
+                quota_us = int(m.group(1))
+            except ValueError:
+                quota_us = None
+            try:
+                period_us = int(m.group(2))
+            except ValueError:
+                period_us = None
+            if quota_us and period_us:
+                cpu_periods_count = 1 / (period_us / 1000000)
+                return int(cpu_periods_count * quota_us)
+        self._log.debug(f'No cgroups2 cpu usage limit set: {content}')
+        return None
+
+    def _get_max_memory_usage(self) -> Optional[int]:
+        max_memory_usage = self._get_cgroups2_max_memory_usage()
+        if not max_memory_usage:
+            # if no cgroups2 limit is in place, then maximum possible memory usage is total available system memory
+            try:
+                max_memory_usage = psutil.virtual_memory().total
+            except KeyError:
+                self._log.warning(f'Unable to determine total available memory', exc_info=True)
+        return max_memory_usage
+
+    def _get_cgroups2_max_memory_usage(self) -> Optional[int]:
+        return self._read_int_from_cgroups('memory.max', 'max memory usage')
+
+
+class LocalMonitorFactory:
+
+    @classmethod
+    def create_local_monitor(cls, parent_logger: Logger, metrics: List[str], engine,
+                             mtab_path=None) -> BaseLocalMonitor:
+        """
+        Constructs concrete instance of BaseLocalMonitor depending on OS and cgroups version.
+        """
+        log = parent_logger.getChild(cls.__name__)
+        if not mtab_path:
+            if is_windows():
+                return StandardLocalMonitor(parent_logger, metrics, engine)
+            mtab_path = '/etc/mtab'
+        cgroups_version, cgroups_fs_path = cls._detect_cgroup_info(log, mtab_path)
+        if cgroups_version == 2:
+            return Cgroups2LocalMonitor(cgroups_fs_path, parent_logger, metrics, engine)
+        elif cgroups_version == 1:
+            # swap for Cgroups1LocalMonitor once it is implemented
+            return StandardLocalMonitor(parent_logger, metrics, engine)
```

Comment:
Cgroups1 looks implemented, shouldn't it be returned here?

Reply:
I made Cgroups1LocalMonitor extend StandardLocalMonitor until it is implemented properly. It will be implemented in a separate PR.